modules/common/job/job.go

```diff
@@ -132,7 +132,7 @@ func (j *Job) DoJob(
 	// Fields outside of the PodTemplate, like TTL does not define what to run
 	// just how to run it. So changing such field should not trigger the re-run
 	// of the Job
-	j.hash, err = util.ObjectHash(j.expectedJob.Spec.Template)
+	j.hash, err = util.ObjectHash(j.expectedJob.Spec.Template.Spec)
```

Comment:
I would extend the comment above and indicate that we ignore the pod metadata as it is also not relevant to what to run.